fix: make account menu width and accounts on account page responsive

rolznz opened the pull request:
### Describe the changes you have made in this PR

- Make the account menu width responsive based on screen size and ensure buttons are not pushed off the screen
- Fix long account name causing a scrollbar in the accounts page

Note: For the account menu this is a bit of a hacky solution because it has to manually account for the menu icon and padding. I couldn't find a better one. Technically this is also an issue in the accounts page, but I did a simpler fix for this one as I think it's less important to have an exact fit. EDIT: it might just be better to fix the account menu width to the same size as the dropdown and remove all those checks 😅 )

### Link this PR to an issue [optional]

Fixes _#1972_

### Type of change

(Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### Screenshots of the changes [optional]
small phone
![image](https://user-images.githubusercontent.com/33993199/212604197-dc814243-3438-489d-aaa9-e959bf4313eb.png)

md breakpoint
![image](https://user-images.githubusercontent.com/33993199/212604345-ca54065b-c7e8-4bb8-a557-e8867d6abf5d.png)

xl screen
![image](https://user-images.githubusercontent.com/33993199/212604383-6e752320-bf77-4e2c-a12c-99677069b2be.png)

Accounts page mobile:
![image](https://user-images.githubusercontent.com/33993199/212605652-708eb5ea-1bbb-48e6-b7a2-2c00a456b7b3.png)

md breakpoint
![image](https://user-images.githubusercontent.com/33993199/212606013-9c733eef-33fd-494c-860d-9ef18a54fa93.png)

xl
![image](https://user-images.githubusercontent.com/33993199/212606123-b9c38ac8-3484-4616-8a44-31c002885e8a.png)


_Add screenshots to make your changes easier to understand. You can also add a video here._

### How has this been tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration_

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

rolznz commented:
I think the dropdown menu width should match the account menu width, what do you think? @reneaaron 

![image](https://user-images.githubusercontent.com/33993199/212606786-9295384e-9724-48c2-ad7b-5e1b406080a9.png)

reneaaron commented:
Nice! 

> I think the dropdown menu width should match the account menu width, what do you think? @reneaaron

Yes. 👍

Can we maybe add a little spacing between the menu icon and the account menu? 

![image](https://user-images.githubusercontent.com/100827540/213312847-f5c4def3-8fc9-4311-9cc3-3ad31a4a8ac4.png)

`mr-4` should be fine 👌

rolznz commented:
> Nice!
> 
> > I think the dropdown menu width should match the account menu width, what do you think? @reneaaron
> 
> Yes. 👍
> 
> Can we maybe add a little spacing between the menu icon and the account menu?
> 
> ![image](https://user-images.githubusercontent.com/100827540/213312847-f5c4def3-8fc9-4311-9cc3-3ad31a4a8ac4.png)
> 
> `mr-4` should be fine 👌

Thanks for testing!

The max-width class wasn't being applied. I've fixed this now so the gap is there.

![image](https://user-images.githubusercontent.com/33993199/213363321-fb4c5f58-6083-4cb6-8fe4-d69f8bc1fe28.png)


Also updated to make the menu width match:
![image](https://user-images.githubusercontent.com/33993199/213363262-0ac00e89-c57b-4410-8d47-00a4360acdc5.png)

escapedcat reviewed (approved):
ack, works great.
Maybe the select should just have a fixed width? Looks a bit weird if it jumps when switching accounts.

e2e is failing on master as well, not related to this PR

rolznz commented:
> ack, works great. Maybe the select should just have a fixed width? Looks a bit weird if it jumps when switching accounts.
> 
> e2e is failing on master as well, not related to this PR

I agree it's weird it changes width when you switch accounts. It would be simpler to just have a fixed width.

reneaaron commented:
@rolznz Do you plan to still change this to a fixed width?

rolznz commented:
@reneaaron I made the change
![image](https://user-images.githubusercontent.com/33993199/214209327-798cd4ec-61d9-47bb-b3de-2efd08d652bf.png)
![image](https://user-images.githubusercontent.com/33993199/214209349-f8673052-2193-43ad-94df-0bf5327f4553.png)

reneaaron commented on src/app/components/Menu/MenuList.tsx:
```diff
@@ -18,11 +20,13 @@ function List({ position = "left", children }: Props) {
-            : "right-0 origin-top-right"
-        } absolute z-50 mt-2 py-1 w-56 rounded-md shadow-lg bg-white ring-1 ring-black ring-opacity-5 focus:outline-none dark:bg-surface-12dp dark:border-0`}
+            : "right-0 origin-top-right",
+          fullWidth ? "w-full" : "w-56",
```
Why do we need a width here?

rolznz replied:
@reneaaron I think to make the menu a consistent width no matter where it is used, and make sure it does not automatically expand if one of the items is wide. I am not sure I understand the question, does this answer it?

reneaaron reviewed (commented):
Tested, feels a lot better now and involves less complexity👌
